test/e2e/persistent_volumes.go

```diff
@@ -327,89 +325,71 @@ func createWaitAndDeletePod(f *framework.Framework, c *client.Client, ns string,
 
-	// Regardless of podErr above, delete the pod if it exists
-	if runPod != nil {
-		err = deletePod(f, c, ns, runPod)
-	}
-
-	// Check results of pod success and pod delete
-	if podErr != nil {
-		errmsg = fmt.Sprintf("Pod %v exited with non-zero exitcode: %v", runPod.Name, podErr)
-	}
-	if err != nil { // Delete error
-		if len(errmsg) > 0 {
-			errmsg += "; and "
-		}
-		errmsg += fmt.Sprintf("Delete error on pod %v: %v", runPod.Name, err)
-	}
-
-	if len(errmsg) > 0 {
-		return fmt.Errorf(errmsg)
-	}
-
-	return nil
+	return
 }
 
-// validate PV/PVC, create and verify writer pod, delete PVC and PV. Ensure
-// all of these steps were successful. Return the pv and pvc to reflect that
-// these resources have been retrieved again (Get).
+// Validate PV/PVC, create and verify writer pod, delete PVC and PV. Ensure that
+// all of these steps were successful.
 // Note: the pv and pvc are returned back to the It() caller so that the
 //   AfterEach func can delete these objects if they are not nil.
-func completeTest(f *framework.Framework, c *client.Client, ns string, pv *api.PersistentVolume, pvc *api.PersistentVolumeClaim) (*api.PersistentVolume, *api.PersistentVolumeClaim, error) {
-
+func completeTest(f *framework.Framework, c *client.Client, ns string, pv *api.PersistentVolume, pvc *api.PersistentVolumeClaim) {
 	// 1. verify that the PV and PVC have binded correctly
 	By("Validating the PV-PVC binding")
-	pv, pvc, err := waitAndValidatePVandPVC(c, ns, pv, pvc)
-	if err != nil {
-		return pv, pvc, err
-	}
+	waitOnPVandPVC(c, ns, pv, pvc)
 
 	// 2. create the nfs writer pod, test if the write was successful,
 	//    then delete the pod and verify that it was deleted
 	By("Checking pod has write access to PersistentVolume")
-	if err = createWaitAndDeletePod(f, c, ns, pvc.Name); err != nil {
-		return pv, pvc, err
-	}
+	createWaitAndDeletePod(f, c, ns, pvc.Name)
 
 	// 3. delete the PVC before deleting PV, wait for PV to be "Available"
 	By("Deleting the PVC to invoke the recycler")
-	pv, pvc, err = deletePVCandValidatePV(c, ns, pvc, pv)
-	if err != nil {
-		return pv, pvc, err
-	}
+	deletePVCandValidatePV(c, ns, pvc, pv)
 
 	// 4. cleanup by deleting the pv
 	By("Deleting the PV")
-	if pv, err = deletePersistentVolume(c, pv); err != nil {
-		return pv, pvc, err
+	deletePersistentVolume(c, pv)
+
+	return
+}
+
+// Validate pairs of PVs and PVCs, create and verify writer pod, delete PVC and validate PV.
+// Ensure each step succeeds.
+func completeMultiTest(f *framework.Framework, c *client.Client, ns string, pvPvcBindMap bindmap) {
+
+	// 1. Verify each PV permits write access to a client pod
+	By("Checking pod has write access to PersistentVolumes")
+	for _, pvc := range pvPvcBindMap {
+		if pvc != nil {
+			createWaitAndDeletePod(f, c, ns, pvc.Name)
+		}
 	}
 
-	return pv, pvc, nil
+	// 2.  Delete each PVC, wait for its bound PV to become "Available"
+	By("Deleting PVCs to invoke recycler")
+	deletePVCandValidatePVGroup(c, ns, pvPvcBindMap)
+
```

Comment:
the PV is allowed to be cleaned up by the AfterEach.  We could explicitly delete them here although the code would be a bit redundant